icu4j/main/classes/core/src/com/ibm/icu/util/TimeZone.java

```diff
@@ -1166,6 +1166,38 @@ public static String getCanonicalID(String id, boolean[] isSystemID) {
         return canonicalID;
     }
 
+    /**
+     * Returns the preferred time zone ID in the IANA database for the given time zone ID.
+     * There are two types of preferred IDs. The first type is the one defined in zone.tab file,
+     * such as "America/Los_Angeles". The second types is the one defined for zones not associated
+     * with a specific region, but not defined with "Link" syntax, such as "Etc/GMT+10".
+     *
+     * <p>Note: For most of system time zone IDs, this method returns an ID same as {@link TimeZone#getCanonicalID(String)}.
+     * {@link TimeZone#getCanonicalID(String)} is based on canonical time zone IDs defined in Unicode CLDR.
+     * These canonical time zone IDs in CLDR were based on very old version of the time zone database.
+     * In the IANA time zone database, some IDs were updated since then. This API returns a newer
+     * time zone ID. For example, CLDR defines "Asia/Calcutta" as the canonical time zone ID. This
+     * method returns "Asia/Kolkata" instead.
+     *
+     * @param id    The input time zone ID.
+     * @return  The preferred time zone ID in the IANA time zone database, or {@link TimeZone#UNKNOWN_ZONE_ID}
+     * if the input ID is not a system ID.
+     * @see #getCanonicalID(String)
+     * @draft ICU 74
+     */
+    public static String getIanaID(String id) {
+        String ianaId = TimeZone.UNKNOWN_ZONE_ID;
+        if (id != null && id.length() != 0) {
```

Comment:
We have a 3 level nested if block here. Could we reduce that by 
```
        if (id == null || id.length() == 0 || id.equals(TimeZone.UNKNOWN_ZONE_ID)) {
            return ianaId;
        }
        String tmpIanaId = ZoneMeta.getIanaID(id);
        if (tmpIanaId == null) {
            return ianaId;
        }
        return tmpIanaId;
```

Reply:
OK

Reply:
done